src/datastation/common/batch_processing.py

```diff
@@ -39,22 +40,32 @@ def __init__(self, wait=0.1, fail_on_first_error=True):
         self.wait = wait
         self.fail_on_first_error = fail_on_first_error
 
-    def process_pids(self, pids, callback):
-        if type(pids) is list:
-            num_pids = len(pids)
+    def process_pids(self, pids_or_objects, callback):
```

Comment:
This is a bit confusing, batchprocessing is already used with callbacks that won't handle dictionaries, so you are now changing the 'contract'. 
I think it is more clear if yo create a separate function for this, so if you only need that PID, you vcan still uste the existing process_pids. For the new case, where we need more, per pid, create a variation on process_pids; maybe process_obj_with_pids ?

Reply:
I gave `process_dicts` a try but had to duplicate too much logic. The backward compatible contract change is expressed in the argument name and docstring. Changing the method name too would touch many other scripts too.